sys/src/9/port/devufs.c

```diff
@@ -138,10 +138,16 @@ ufsread(Chan *c, void *a, int32_t n, int64_t offset)
 	return n;
 }
 
+static int32_t
+chanread(struct mount* mp, void *p, int32_t n, int64_t off)
```

Comment:
I don't get why you need this. Why isn't dev->read enough? 

FYI the Plan 9 way to do this is you store the type in the QID and case out on that in dev->read.